File: app/src/main/java/com/bpn/auto/core/Logger.java
package com.bpn.auto.core;

import android.annotation.SuppressLint;
import java.text.SimpleDateFormat;
import java.util.Date;

@SuppressLint("SimpleDateFormat")
public class Logger {

  public enum LogLevel {
    DEBUG,
    INFO,
    WARN,
    ERROR,
    NONE
  }

  private static final String GREEN = "\033[92m";
  private static final String RED = "\033[91m";
  private static final String BLUE = "\033[94m";
  private static final String YELLOW = "\033[93m";
  private static final String RESET = "\033[0m";
  private static final SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS");

  private final LogLevel logLevel;

  public Logger() {
    this.logLevel = LogLevel.INFO;
  }

  public Logger(LogLevel level) {
    this.logLevel = level;
  }

  private static String getCurrentTimestamp() {
    return formatter.format(new Date());
  }

  public void d(String msg) {
    if (logLevel.ordinal() <= LogLevel.DEBUG.ordinal()) {
      String time = getCurrentTimestamp();
      String message = String.format("[%s] %s[DEBUG]%s %s", time, BLUE, RESET, msg);
      System.out.println(message);
    }
  }

  public void i(String msg) {
    if (logLevel.ordinal() <= LogLevel.INFO.ordinal()) {
      String time = getCurrentTimestamp();
      String message = String.format("[%s] %s[INFO]%s %s", time, GREEN, RESET, msg);
      System.out.println(message);
    }
  }

  public void w(String msg) {
    if (logLevel.ordinal() <= LogLevel.WARN.ordinal()) {
      String time = getCurrentTimestamp();
      String message = String.format("[%s] %s[WARN]%s %s", time, YELLOW, RESET, msg);
      System.out.println(message);
    }
  }

  public void e(String msg) {
    if (logLevel.ordinal() <= LogLevel.ERROR.ordinal()) {
      String time = getCurrentTimestamp();
      String message = String.format("[%s] %s[ERROR]%s %s", time, RED, RESET, msg);
      System.out.println(message);
    }
  }
}

File: steam/app/logger.py
from enum import Enum
from datetime import datetime


class LogLevel(Enum):
  DEBUG = 0
  INFO = 1
  WARN = 2
  ERROR = 3
  NONE = 4


class Logger:
  GREEN = "\033[92m"
  RED = "\033[91m"
  BLUE = "\033[94m"
  YELLOW = "\033[93m"
  RESET = "\033[0m"

  def __init__(self, level=LogLevel.INFO):
    self.log_level = level

  @staticmethod
  def get_current_timestamp():
    return datetime.now().strftime("%Y-%m-%d %H:%M:%S.%f")[:-3]

  def d(self, msg):
    if self.log_level.value <= LogLevel.DEBUG.value:
      time = self.get_current_timestamp()
      message = f"[{time}] {self.BLUE}[DEBUG]{self.RESET} {msg}"
      print(message)

  def i(self, msg):
    if self.log_level.value <= LogLevel.INFO.value:
      time = self.get_current_timestamp()
      message = f"[{time}] {self.GREEN}[INFO]{self.RESET} {msg}"
      print(message)

  def w(self, msg):
    if self.log_level.value <= LogLevel.WARN.value:
      time = self.get_current_timestamp()
      message = f"[{time}] {self.YELLOW}[WARN]{self.RESET} {msg}"
      print(message)

  def e(self, msg):
    if self.log_level.value <= LogLevel.ERROR.value:
      time = self.get_current_timestamp()
      message = f"[{time}] {self.RED}[ERROR]{self.RESET} {msg}"
      print(message)
